src/sentry/models/organization.py

```diff
@@ -272,6 +290,46 @@ def update(self, *args, **kwargs):
     def reserve_snowflake_id(cls):
         return generate_snowflake_id(cls.snowflake_redis_key)
 
+    # Used to generate Organization specific slugs with a region prefix if one
+    # is required. If the slug is prefixed with any character combination
+    # resulting in it starting with `r-XX-`, this sequence will be truncated
+    # by default in order to prevent a collision with any future regions.
+    @classmethod
+    def generate_regional_slug(
+        cls, slugify_target: str, truncate_region_prefix_collisions: bool = True
+    ) -> str:
+        sanitized_slug_target = slugify_target.lower().replace("_", "-").strip("-")
+        starts_with_region_prefix = match(region_prefix_regex, sanitized_slug_target)
+
+        if starts_with_region_prefix and truncate_region_prefix_collisions:
+            region_mod_split = split(region_prefix_regex, sanitized_slug_target)
```

Comment:
```suggestion
            region_mod_split = region_prefix_regex.split(sanitized_slug_target)
```

To go with the pre-compile suggestion.